content/go/server.go.md

```diff
@@ -0,0 +1,522 @@
+
+	// Call the getHardware() method to get list of servers that matches with the filter
+	hardware, err := accountService.Mask(mask).Filter(filter).GetHardware()
+	if err != nil {
+		fmt.Printf("\n Unable to find server '"+hostname+"'\n - %s\n", err)
+		return
+	}
+
+	// If server name was not found we throw a message
+	if len(hardware) < 1 {
+		fmt.Printf("\n Unable to find server '%s'\n", hostname)
+	}
+	return *hardware[0].Id
+}
+
+/**
+Prints server details
+*/
+func printServerDetail(server datatypes.Hardware_Server) {
+	rows := []string{"Name", "Value"}
+	cmd := terminal.NewStdUI()
+	table := cmd.Table(rows)
+	table.Add("ID", fmt.Sprintf("%d", *server.Id))
+	table.Add("GUID", sl.Get(server.GlobalIdentifier).(string))
+	table.Add("Hostname", sl.Get(server.Hostname).(string))
+	table.Add("Domain", sl.Get(server.Domain).(string))
+	table.Add("FQDN", sl.Get(server.FullyQualifiedDomainName).(string))
+	if server.HardwareStatus != nil {
+		table.Add("Status", sl.Get(server.HardwareStatus.Status).(string))
+	}
+	if server.Datacenter != nil {
+		table.Add("Datacenter", sl.Get(server.Datacenter.Name).(string))
+	}
+	table.Add("CPU cores", fmt.Sprintf("%d", *server.ProcessorPhysicalCoreAmount))
+	table.Add("Memory", (fmt.Sprintf("%d", *server.MemoryCapacity) + "G"))
+	table.Add("Public IP", sl.Get(server.PrimaryIpAddress).(string))
+	table.Add("Private IP", sl.Get(server.PrimaryBackendIpAddress).(string))
+	table.Add("IPMI IP", sl.Get(server.NetworkManagementIpAddress).(string))
+	if server.OperatingSystem != nil &&
+		server.OperatingSystem.SoftwareLicense != nil &&
+		server.OperatingSystem.SoftwareLicense.SoftwareDescription != nil {
+		table.Add("OS", sl.Get(server.OperatingSystem.SoftwareLicense.SoftwareDescription.Name).(string))
+		table.Add("OS version", sl.Get(server.OperatingSystem.SoftwareLicense.SoftwareDescription.Version).(string))
+	}
+	if server.Notes != nil && *server.Notes != "" {
+		table.Add("Note", sl.Get(server.Notes).(string))
+	}
+	cmd.Say("Server details")
+	table.Print()
+}
+
+/**
+Prints the data as format JSON.
+*/
+func printAsJsonFormat(data interface{}) {
+	jsonData, jsonErr := json.MarshalIndent(data, "", "    ")
+	if jsonErr != nil {
+		fmt.Println(jsonErr)
+		return
+	}
+	println(string(jsonData))
+}
+
+/*
+Cancel a bare metal server
+
+This function looks for a server by hostname to cancel it. The server will be cancelled immediately if
+it is billed Hourly, for Monthly server the cancellation will be made after next bill date.
+*/
+
+func cancelServer(serverId int) {
+
+	// In order to cancel billing item of server we need to set following values
+	cancelAssociatedBillingItems := false
+	reason := "No longer needed"
+	customerNote := "BMS - canceled"
+
+	// Get SoftLayer_Hardware_Server, and SoftLayer_Billing_Item services
+	hardwareService := services.GetHardwareServerService(sess)
+	billingService := services.GetBillingItemService(sess)
+	mask := "id;hostname;billingItem;hourlyBillingFlag"
```

Comment:
mask := "id; hostname; billingItem; hourlyBillingFlag"

Reply:
when I added a space after `;` request brokes, returning a `Response: {"error":"Bad Request","code":"SoftLayer Exception Public"}`, not sure why, so I think I'll keep the initial syntax in this case.